framework/caching/DbQueryDependency.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\caching;
+
+use yii\base\InvalidConfigException;
+use yii\db\Connection;
+use yii\db\QueryInterface;
+use yii\di\Instance;
+
+/**
+ * DbQueryDependency represents a dependency based on the query result of an [[QueryInterface]] instance.
+ *
+ * If the query result changes, the dependency is considered as changed.
+ * The query is specified via the [[query]] property.
+ *
+ * Object of any class which matches [[QueryInterface]] can be used, so this dependency can be used not only
+ * with regular relational databases but with MongoDB, Redis and so on as well.
+ *
+ * For more details and usage information on Cache, see the [guide article on caching](guide:caching-overview).
+ *
+ * @see QueryInterface
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @since 2.0.12
+ */
+class DbQueryDependency extends Dependency
+{
+    /**
+     * @var string|array|object the application component ID of the database connection, connection object or
+     * its array configuration.
+     * This field can be left blank, allowing query to determine connection automatically.
+     */
+    public $db;
+    /**
+     * @var QueryInterface the query which result is used to determine if the dependency has been changed.
+     * Actual query method to be invoked is determined by [[method]].
+     */
+    public $query;
+    /**
+     * @var string|callable method which should be invoked in over the [[query]] object.
+     *
+     * If specified as a string an own query method with such name will be invoked, passing [[db]] value as its
+     * first argument. For example: `exists`, `all`.
+     *
+     * This field can be specified as a PHP callback of following signature:
+     *
+     * ```php
+     * function (QueryInterface $query, mixed $db) {
+     *     //return mixed;
+     * }
+     * ```
+     *
+     * If not set - [[QueryInterface::one()]] will be used.
+     */
+    public $method;
+
+
+    /**
+     * Generates the data needed to determine if dependency is changed.
+     * This method returns the query result
+     * @param Cache $cache the cache component that is currently evaluating this dependency
+     * @return mixed the data needed to determine if dependency has been changed.
+     * @throws InvalidConfigException on invalid configuration.
+     */
+    protected function generateDependencyData($cache)
+    {
+        $db = $this->db;
+        if ($db !== null) {
+            $db = Instance::ensure($db);
+        }
+
+        if (!$this->query instanceof QueryInterface) {
+            throw new InvalidConfigException('`' . get_class($this) . '::$query` should be an instance of `yii\db\QueryInterface`.');
```

Comment:
Since errors aren't outputted with markdown processing it's better to use regular " instead of `.